app/actions/handlers.py

```diff
@@ -155,49 +155,45 @@ def fn(p): return getattr(p, device_state.recorded_at_field_name)
                 )
 
                 if transformed_data:
-                    def generate_batches(iterable, n=action_config.observations_per_request):
-                        for i in range(0, len(iterable), n):
-                            yield iterable[i: i + n]
-
-                    for i, batch in enumerate(generate_batches(transformed_data)):
-                        async for attempt in stamina.retry_context(
-                                on=httpx.HTTPError,
-                                attempts=3,
-                                wait_initial=datetime.timedelta(seconds=10),
-                                wait_max=datetime.timedelta(seconds=10),
-                        ):
-                            with attempt:
-                                try:
-                                    logger.info(
-                                        f'Sending observations batch #{i}: {len(batch)} observations. Device {device_id}'
-                                    )
-                                    await send_observations_to_gundi(
-                                        observations=batch,
-                                        integration_id=str(integration.id)
-                                    )
-                                except httpx.HTTPError as e:
-                                    msg = f'Sensors API returned error for integration_id: {str(integration.id)}. Exception: {e}'
-                                    logger.exception(
-                                        msg,
-                                        extra={
-                                            'needs_attention': True,
-                                            'integration_id': str(integration.id),
-                                            'action_id': "pull_observations"
-                                        }
-                                    )
-                                    raise e
-                                else:
-                                    for vehicle in batch:
-                                        # Update state
-                                        state = {
-                                            "recorded_at": vehicle.get("recorded_at")
-                                        }
-                                        await state_manager.set_state(
-                                            str(integration.id),
-                                            "pull_observations",
-                                            state,
-                                            vehicle.get("source")
-                                        )
+                    # We send only the latest 100 obs
+                    transformed_data = sorted(transformed_data[:100], key=lambda x: x.get("recorded_at"), reverse=True)
```

Comment:
I know we don't want to expose the setting to users, but what do you think about reading it from an env var in settings with 100 as default? That way we can change it later if we see any issue in prod, without needing code changes.